File: counting-sort/Python/counting_sort.py
def counting_sort(numbers):
    tally = {}
    sorted = []
    start = 1
    end = len(numbers) + 1

    for num in numbers:
        value = tally.get(num)
        if value != None:
            tally[num] = value + 1
        else:
            tally[num] = 1

    for i in range(start, end):
        value = tally.get(i)
        if value != None:
            for j in range(1, value + 1):
                sorted.append(i)

    return sorted


numbers = [14,18,18,17,7,13,8,13,3,4,12,8,3,11,14,15,18,15,9,10]

print(counting_sort(numbers))

File: counting-sort/CSharp/CountingSort/CountingSort/CountingSort.cs
using System.Collections.Generic;

namespace CountingSort
{
    public static class CountingSort
    {
        public static int[] Sort(int[] numbers)
        {
            var tally = new Dictionary<int, int>();
            var sorted = new List<int>();

            foreach(var value in numbers)
            {
                if (tally.ContainsKey(value))
                {
                    var sum = tally[value];
                    tally[value] = sum + 1;
                }
                else
                {
                    tally[value] = 1;
                }
            }
            
            for (var i = 1; i <= numbers.Length; i++)
            {
                if (tally.ContainsKey(i))
                {
                    var values = tally[i];
                    for (var j = 0; j < values; j++)
                    {
                        sorted.Add(i);
                    }
                }
            }

            return sorted.ToArray();
        }
    }
}
